cloud/pkg/admissioncontroller/controller/admission.go

```diff
@@ -0,0 +1,255 @@
+			Namespace: constants.NamespaceName,
+			Name:      constants.ServiceName,
+		},
+		Spec: v1.ServiceSpec{
+			Type: "ClusterIP",
+			Ports: []v1.ServicePort{
+				{
+					Port:       constants.Port,
+					TargetPort: intstr.FromInt(constants.Port),
+				},
+			},
+		},
+	}
+	_, err := ac.Client.CoreV1().Services(constants.NamespaceName).Create(service)
+	if err != nil {
+		klog.Warningf("Failed to create webhook service with error: %s", err)
+	}
+	//create Endpoints
+	endpoint := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: constants.ServiceName,
+		},
+		Subsets: []v1.EndpointSubset{
+			{
+				Addresses: []v1.EndpointAddress{
+					{
+						IP: localIP,
+					},
+				},
+				Ports: []v1.EndpointPort{
+					{
+						Port: constants.Port,
+					},
+				},
+			},
+		},
+	}
+	_, err = ac.Client.CoreV1().Endpoints(constants.NamespaceName).Create(endpoint)
+	if err != nil {
+		klog.Warningf("Failed to create endpoints with error: %s", err)
+	}
+}
+
+func (ac *AdmissionController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	// verify the content type is accurate
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		klog.Errorf("contentType=%s, expect application/json", contentType)
+		return
+	}
+
+	klog.Infof("handling request: %v", body)
+
+	// The AdmissionReview that was sent to the webhook
+	requestedAdmissionReview := admissionv1beta1.AdmissionReview{}
+
+	// The AdmissionReview that will be returned
+	responseAdmissionReview := admissionv1beta1.AdmissionReview{}
+
+	deserializer := codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
+		klog.Error(err)
+		responseAdmissionReview.Response = toAdmissionResponse(err)
+	} else {
+		responseAdmissionReview.Response = ac.admit(requestedAdmissionReview)
+	}
+
+	// Return the same UID
+	responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+
+	klog.Infof("sending response: %v", responseAdmissionReview.Response)
+
+	respBytes, err := json.Marshal(responseAdmissionReview)
+	if err != nil {
+		klog.Error(err)
+	}
+	if _, err := w.Write(respBytes); err != nil {
+		klog.Error(err)
+	}
+}
+
+func (ac *AdmissionController) admit(review admissionv1beta1.AdmissionReview) *admissionv1beta1.AdmissionResponse {
+	reviewResponse := admissionv1beta1.AdmissionResponse{}
+	reviewResponse.Allowed = true
+
+	var msg string
+	switch review.Request.Operation {
+	case admissionv1beta1.Create, admissionv1beta1.Update, admissionv1beta1.Delete:
+		//TODO: abnormal configuration will be detected here, so far, not allowed to create it!
+		reviewResponse.Allowed = false
+		msg = msg + "the config holds unwanted key or value"
+	default:
+		klog.Warningf("Unsupported webhook operation %v", review.Request.Operation)
+		reviewResponse.Allowed = true
```

Comment:
we can't allow any other operation since it is not supported right ? why is this flag set to true ?

Reply:
@rohitsardesai83 , so far, those are all operations are available from `admissionv1beta1`.  
and set the to true since this patch doesn't give any real validation, the real validation will be done in another PR, this PR hopefully is a framework.